api/src/main/java/org/hawkular/apm/api/utils/TimeUtil.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.apm.api.utils;
+
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This class provides time utility functions.
+ *
+ * @author gbrown
+ */
+public class TimeUtil {
+
+    /**
+     * This method returns the timestamp, associated with the supplied
+     * instant, in microseconds.
+     *
+     * @param instant The instant
+     * @return The timestamp in microseconds
+     */
+    public static long toMicros(Instant instant) {
+        return TimeUnit.SECONDS.toMicros(instant.getEpochSecond()) + TimeUnit.NANOSECONDS.toMicros(instant.getNano());
```

Comment:
I'm not sure if this can give us good micro seconds precision. Instant underneath uses `System.currentTimeMillis()` (`Instant.now()`).So this will result in current time millis + some nano seconds which basically is duration of  additional operations.